src/WPF/ArcGISRuntime.WPF.Viewer/Samples/GraphicsOverlay/SketchOnMap/SketchOnMap.xaml.cs

```diff
@@ -133,16 +138,27 @@ private async Task<Graphic> GetGraphicAsync()
         }
         #endregion
 
-        private async void DrawButtonClick(object sender, RoutedEventArgs e)
+        private void ShapeClick(object sender, RoutedEventArgs e)
+        {
+            // Get the command parameter from the button press
+            string mode = (sender as Button).CommandParameter.ToString();
+
+            // Check if the command parameter is defined in the SketchCreationMode enumerator
+            if (Enum.IsDefined(typeof(SketchCreationMode), mode))
+            {
+                _ = DrawButtonClick((SketchCreationMode)Enum.Parse(typeof(SketchCreationMode), mode));
+            }
+        }
+
+        private async Task DrawButtonClick(SketchCreationMode creationMode)
```

Comment:
I recommend renaming this method.
```suggestion
        private async Task StartSketch(SketchCreationMode creationMode)
```